plugins/cache/redis.py

```diff
@@ -217,13 +217,13 @@ def delete(self, key):
         self._db.zrem(self._keys_set, key)
 
     def flush(self):
-        for key in self.keys():
+        for key in list(self.keys()):
             self.delete(key)
 
     def copy(self):
         # TODO: there is probably a better way to do this in redis
         ret = dict()
-        for key in self.keys():
+        for key in list(self.keys()):
```

Comment:
```suggestion
        for key in self.keys():
```
This only makes it less efficient both on Python 2 and Python 3.